### PR TITLE
Fix #12: create static site for documentation using Docusaurus

### LINKED ISSUE #12: Create a docusaurus site for /blog and /docs
We need an easy way to create documentation, build community via blogging, etc. Let's try using https://docusaurus.io/ to build this as static content.

### PR DESCRIPTION
**Pull request by mmBabol**
Created basic [Docusaurus](https://docusaurus.io/) template containing all our ReadMe information, will be updated as we progress with the project.

To test this out, pull my branch and do the following:

```
npm install
npm run docs
```

Here is what it looks like when running on my box:

screen shot 2018-05-16 at 12 59 36 pm

### CODE REVIEW

**Comment by humphd** [comment]
I tried this, and I'm getting the following:

```bash
~/repos/next (docusaurus) $ npm run docs

> unbundled@0.0.1 docs /Users/humphd/repos/next
> cd website && npm start


> @ start /Users/humphd/repos/next/website
> docusaurus-start

sh: docusaurus-start: command not found
npm ERR! file sh
npm ERR! code ELIFECYCLE
npm ERR! errno ENOENT
npm ERR! syscall spawn
npm ERR! @ start: `docusaurus-start`
npm ERR! spawn ENOENT
npm ERR!
npm ERR! Failed at the @ start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm WARN Local package.json exists, but node_modules missing, did you mean to install?

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/humphd/.npm/_logs/2018-05-16T16_39_00_965Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! unbundled@0.0.1 docs: `cd website && npm start`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the unbundled@0.0.1 docs script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/humphd/.npm/_logs/2018-05-16T16_39_00_990Z-debug.log
```

**Comment by humphd** [comment]
I was able to solve my install problem by adding `docusaurus` to the root `package.json` file:

```diff
diff --git a/package.json b/package.json
index 0ed0965..465c8fa 100644
--- a/package.json
+++ b/package.json
@@ -19,6 +19,7 @@
   },
   "homepage": "https://github.com/humphd/next#readme",
   "devDependencies": {
+    "docusaurus": "^1.0.15",
     "husky": "^0.14.3",
     "precise-commits": "^1.0.2",
     "prettier": "^1.12.1"
```

File: .gitignore-example-from-docusaurus [hunk 1]
```diff
@@ -0,0 +1,13 @@
+.DS_Store
```
Comment (Fatehsandhu) [inline .gitignore-example-from-docusaurus]: Hey @mmBabol Can you please merge this file with the one for our project. Also, remove this file kindly.
Thanks

File: docs/doc1.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
+---
+id: doc1
```
Comment (humphd) [inline docs/doc1.md]: Is this name important?  It seems odd to use `doc1` since we'll have lots of them.

File: docs/doc2.md [hunk 3]
```diff
@@ -0,0 +1,178 @@
+---
+id: doc2
+title: Architecture
```
Comment (humphd) [inline docs/doc2.md]: For example, why not name this `architecture.md`?

File: website/package.json [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+    "examples": "docusaurus-examples",
+    "start": "docusaurus-start",
+    "build": "docusaurus-build",
+    "publish-gh-pages": "docusaurus-publish",
+    "write-translations": "docusaurus-write-translations",
+    "version": "docusaurus-version",
+    "rename-version": "docusaurus-rename-version"
+  },
+  "devDependencies": {
+    "docusaurus": "^1.0.15"
```
Comment (humphd) [inline website/package.json]: I'm not sure we need this here if we're doing it in our root `package.json` file instead.

**Comment by humphd** [comment]
You can remove `website/yarn.lock` after you move the dependencies to our root `package.json`.

**Comment by humphd** [comment]
We need to add some docs to our README or somewhere to tell us how to rebuild this site.

Also, we should hook this into our `.travis.yml` file.  We could to that in a new bug if you like, but we should file that.

**Pull request closed without merging**